authz/api/__init__.py

```diff
@@ -1,77 +1,143 @@
-from flask import Blueprint, jsonify, abort, request
-from sqlalchemy.orm.exc import NoResultFound
+from flask import Blueprint, jsonify, abort, request, Response
 
-from authz.models import Role, User, WorkspaceRole
 from authz.serializers.role import RoleSerializer
+from authz.serializers.user import UserSerializer
+from authz.serializers.workspace_user import WorkspaceUserSerializer
 
-api = Blueprint('api', __name__)
+from authz.domain.exceptions import NotFoundError, AlreadyExistsError
+from authz.domain.workspace_users import WorkspaceUsers
+from authz.domain.users import Users
+from authz.domain.roles import Roles
 
+api = Blueprint("api", __name__)
 
-@api.route('/roles')
+
+@api.route("/roles")
 def get_roles():
-    roles = Role.query.all()
+    """
+    Returns a list of all existing roles.
+
+    GET /roles
+    """
+
+    roles = Roles.get_all()
     return RoleSerializer().jsonify(roles, many=True)
 
 
-@api.route('/roles/<string:name>')
+@api.route("/roles/<string:name>")
 def get_role(name):
+    """
+    Returns a given role.
+
+    GET /roles/<role name>
+    """
+
     try:
-        role = Role.query.filter_by(name=name).one()
-    except NoResultFound:
-        return abort(404)
+        role = Roles.get(name)
+    except NotFoundError as e:
+        return (jsonify({"error": e.message}), 404)
+
     return RoleSerializer().jsonify(role)
 
 
-@api.route('/workspaces/<uuid:workspace_id>/users', methods=['PUT'])
+@api.route("/users", methods=["POST"])
+def create_user():
+    """
+    Creates a new atat user with the given role.
+
+    POST /users/
+    {
+        id: <user id>,
+        atat_role: <name of an atat role>
+    }
+
+    Returns the created user.
+    """
+
+    try:
+        new_user_dict = request.json
+        user_id = new_user_dict["id"]
+        user_atat_role = new_user_dict["atat_role"]
```

Comment:
Can we leverage `UserSerializer.load(request.json)` to handle the input validation in these routes?

Reply:
We probably could, but I can't really figure out how to do it in a satisfactory way. I don't really want to leverage Marshmallow to create model objects, and I don't want to tie the input schema to the output schema. If you think it's ok, I may just keep doing the input validation manually for now since the request bodies are pretty simple.